### ISSUE TITLE
API-posted user links show up with @ symbol stripped in hubot-slack message?

### ISSUE THREAD

**Issue by jbz** [slackapi/hubot-slack#333]
- [x] I've read and understood the [Contributing guidelines](./CONTRIBUTING.md) and have done my best effort to follow them.
- [x] I've read and agree to the [Code of Conduct](./CODE_OF_CONDUCT.md).
- [x] I've searched for any related issues and avoided creating a duplicate issue.
#### Description

I'm trying to have a python script trigger a hubot action via Slack.  As part of my hubot script, I accept a command with a list of Slack users as arguments, i.e. "@foo @bar @baz".  The regex on the script then looks for args in the match with starting @ symbols to determine it's been given valid usernames.   The problem is that when my python script calls Slack via the Slack API and submits the @foo @bar @baz, as far as I can tell the Slack API helpfully converts it into user links.  In the slack client I see them as @foo @bar @baz with links to the users.  But in the debug output of my hubot, I see it as foo bar baz.  I presume this is because the formatter in the connector is stripping the link information out.  Ugh.  Is there any way to identify user links (as distinct from just the text of the username) using the hubot-slack adapter when it's part of the message body? I swear I've looked up and down Google, and I think I'm just not quite tech savvy enough to figure this out via the code.  I've read the formatter but I'm just not grokking.  Any tips massively appreciated.  I do apologize for putting this in a github issue.
#### Reproducible in:

{project_name} version:
OS version(s):
Device(s):
#### Steps to reproduce:
1. Submit message to Slack channel via API so usernames in message body (@foo) are converted to user links (<@U12345678|foo>).
2. Examine incoming Slack messages via hubot slack connector (regex or debug)
3. Note that the usernames have had the link (and @) stripped, making it impossible to ID them as usernames.
### Expected result:

 I expected username links in slack inside a message body to be identifiable from hubot-slack scripts programatically
### Actual result:

 I don't understand how to do that.
### Attachments:

Here's the log as the slack client sees it:

![in slack client](https://cloud.githubusercontent.com/assets/243143/17231914/c59b9fde-54f2-11e6-88bd-1c68ebe71a34.png)

Here's the log as the hubot receives it:

```
[Thu Jul 28 2016 18:36:14 GMT-0400 (EDT)] DEBUG Executing listener callback for Message 'marvin rollcall start ian.miller lezou bender mako '
[Thu Jul 28 2016 18:36:14 GMT-0400 (EDT)] DEBUG Sending to test-channel: Starting a rollcall at 07/28 18:36 for the following attendees:
[Thu Jul 28 2016 18:36:14 GMT-0400 (EDT)] DEBUG Message 'marvin rollcall start ian.miller lezou bender mako ' matched regex //(.*)/i/
[Thu Jul 28 2016 18:36:14 GMT-0400 (EDT)] DEBUG Executing listener callback for Message 'marvin rollcall start ian.miller lezou bender mako '
```

**Comment by bar** [comment]
@jbz hmm, was I invoked to a handle huddle?

**Comment by jbz** [comment]
Oh lord, @bar I'm so sorry.  I should have been more careful about my example text.  My apologies.  I completely forgot it would long on github too :-/

**Comment by bar** [comment]
haha no worries, this is always funny

**Comment by DEGoodmanWilson** [comment]
😂

**Comment by DEGoodmanWilson** [comment]
OK, will put this on the stack, thanks for raising it!

**Comment by jbz** [comment]
Humbly grateful thanks.  Currently looking at having my script validate potential usernames out of band using the Slack API.

**Comment by iancward** [comment]
I think this may have been fixed in 4.0.4..

**Comment by DEGoodmanWilson** [comment]
@jbz can you confirm?

**Comment by jbz** [comment]
This is not fixed as of 4.1.0.  I still see my hubot script (hubot-rollcall) responding to a call made in slack to start the rollcall, and not seeing the usernames that are part of the call:


shamu rollcall start @<username> @<username>

ShamuBOT [12:45 PM]  
If you're subbing for someone, please say `shamu sub for @<username>`.

[12:45]  
If you're here, please say `shamu here`.

[12:45]  
Starting a rollcall at 10/11 16:45 for the following attendees:

...there are no attendees, because my script checks for input text in the form of words prefaced by '@' to identify users.  There are no such users, because the text is stripped of the user symbol prefix. :-(

**Comment by iancward** [comment]
I just upgraded hubot-slack to 4.3.3 and it looks like the `@` character is now preserved in mentions; however, what I'm really looking for is the ability to keep the UID in the message (or somehow get it out of the message when a user is @ mentioned. This is primarily so that hubot can make an API call to get a user's profile, in order to get the e-mail address of the user.

Right now it doesn't seem possible to do this with how inbound messages are formatted.
